stripe/src/main/java/com/stripe/android/paymentsheet/BuyButton.kt

```diff
@@ -109,6 +112,20 @@ internal class BuyButton @JvmOverloads constructor(
         }
     }
 
+    fun updateState(state: ViewState) {
+        when (state) {
+            is ViewState.Ready -> {
```

Comment:
you don't need the `is` here (and in fact line 120 doesn't have it?)

Reply:
I need it because `ViewState.Ready` is a class, while `ViewState.Confirming` is an object